### PR TITLE
ICU-22900 Remove deprecated SimpleNumber truncateStart

### PR DESCRIPTION
**Pull request by sffc**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22900
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [x] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Comment by markusicu** [comment]
> This should ship in ICU 77

I don't see why this shouldn't go into 76. We are not removing anything that has ever been stable. The two functions that are to be removed went from draft ICU 73 to deprecated ICU 75, so anyone limiting themselves to stable API has never used them.

File: icu4c/source/i18n/unicode/simplenumberformatter.h [hunk 1]
```diff
@@ -78,24 +78,12 @@ class U_I18N_API SimpleNumber : public UMemory {
      */
     void roundTo(int32_t power, UNumberFormatRoundingMode roundingMode, UErrorCode& status);
 
-#ifndef U_HIDE_DRAFT_API
     /**
      * Sets the number of integer digits to the given amount, truncating if necessary.
      *
-     * @draft ICU 75
+     * @stable ICU 75
```
Comment (markusicu) [inline icu4c/source/i18n/unicode/simplenumberformatter.h:84]: We agreed to remove truncateStart() in ICU 76. I think we should in fact do so.
Of course, we should not promote this replacement to stable in 76.
Even if we pushed removing truncateStart() to 77, we should leave the promotion to the usual BRS task.

So please don't do the promotion in this PR.

File: icu4c/source/i18n/unicode/usimplenumberformatter.h [hunk 2]
```diff
@@ -156,27 +156,13 @@ U_CAPI void U_EXPORT2
 usnum_setMinimumFractionDigits(USimpleNumber* unumber, int32_t minimumFractionDigits, UErrorCode* ec);
 
 
-#ifndef U_HIDE_DRAFT_API
 /**
  * Sets the number of integer digits to the given amount, truncating if necessary.
  *
- * @draft ICU 75
+ * @stable ICU 75
```
Comment (markusicu) [inline icu4c/source/i18n/unicode/usimplenumberformatter.h:162]: ditto

**Review by richgillam — COMMENTED** [review]
I think Markus is right that we shouldn't jump the gun on promoting anything or do it in a special PR.  I can take care of that the way I always do.

I'm agnostic on whether we should delete these APIs now or in 77, but I'm leaning toward Markus's position.  The longer they're in there, the longer we have for people to start using them, making it worse when we get rid of them.  And they're already marked as deprecated.  I'm not sure I see the value in keeping them around.

File: icu4c/source/i18n/unicode/simplenumberformatter.h (hunk 1 above)
Comment (sffc) [reply icu4c/source/i18n/unicode/simplenumberformatter.h]: > Even if we pushed removing truncateStart() to 77, we should leave the promotion to the usual BRS task.

Ok that's reasonable

**Review by markusicu — APPROVED** [review]
Thanks -- please merge this today.

**Pull request merged**